### ISSUE TITLE
GRACE-1 and GRACE-2 TLE returning NaN's

### ISSUE THREAD

**Issue by gohde** [brandon-rhodes/python-sgp4#21]
Hi,
the Grace-1 and Grace-2 TLE are returning NaN's.

`
from sgp4.earth_gravity import wgs72,wgs72old,wgs84
from sgp4.io import twoline2rv
import sgp4

line1 = ('1 27392U 02012B   17317.13935224  .00120532  42499-5  34060-3 0  9998')
line2 = ('2 27392  88.9800 298.9359 0008907 105.9560 254.2707 15.93313173881025')
satellite = twoline2rv(line1, line2, wgs72)
position, velocity = satellite.propagate(2000, 6, 29, 12, 50, 20.1)

print(position)
print(velocity)
print(satellite.error_message)

>>> (nan, nan, nan)
>>> (nan, nan, nan)
>>> mean eccentricity -0.025182 not within range 0.0 <= e < 1.0
`

I confirmed the TLE from multiple sources online. I'm using this for a 3d visualisation and am very much out of my depth as to what is going on under the hood to figure out where the error could originate.

Best regards,
Fabian

**Comment by kartikkumar** [comment]
So from Space-Track.org, these are the latest 3-line TLEs for GRACE-1 and GRACE-2

```
0 GRACE 1
1 27391U 02012A   17317.12403959  .00028171 -75101-6  98164-4 0  9991
2 27391  88.9806 298.7903 0008579 101.0296 315.7217 15.89038651881020
0 GRACE 2
1 27392U 02012B   17317.45338623  .00133498  54732-5  37547-3 0  9994
2 27392  88.9798 298.8885 0008928 105.6680 254.5590 15.93408025881076
```

Line 2 contains the eccentricity, which is `0.0008579` and `0.0008928` respectively, which is similar to yours.

You're trying to propagate to a date in 2000. The GRACE mission launched on 17 March 2002, which means they weren't even in orbit! Moreover, depending on the epoch of the TLE that you quoted, the orbital elements after propagation are valid for a few weeks, to a few months at most (the solution degrades quite quickly).

In short, it looks like you're trying something that the TLEs are not intended for, i.e., propagation over long time periods. This would explain the error, which shows that the eccentricity becomes negative, which is non-physical. 

Your best bet is to convert the TLE to Cartesian elements/orbital elements by propagating to the epoch of the TLE, and then using a higher fidelity propagator that can be used for longer time scales. I would also question why you're trying to propagate to the year 2000, before the mission even launched.

**Comment by gohde** [comment]
Hi, thank you so much and duh, I completely oversaw the date, that was due for replacement later with something from this month. Basically I'm doing a 3d animation that needed a specific set of satellites going about their business. Now everything works and I can render :-) Again, thank you and please excuse my ignorance.

**Comment by kartikkumar** [comment]
Glad you got it fixed! 👍
